app/src/main/java/com/termux/app/terminal/TermuxTerminalViewClient.java

```diff
@@ -410,10 +412,15 @@ public void onToggleSoftKeyboardRequest() {
                 mActivity.getPreferences().setSoftKeyboardEnabled(false);
                 KeyboardUtils.disableSoftKeyboard(mActivity, mActivity.getTerminalView());
             } else {
+                // Show with a delay, otherwise pressing keyboard toggle won't show the keyboard after
+                // switching back from another app if keyboard was previously disabled by user.
+                // Also request focus, since it wouldn't have been requested at startup by
+                // setSoftKeyboardState if keyboard was disabled. #2112
                 Logger.logVerbose(LOG_TAG, "Enabling soft keyboard on toggle");
                 mActivity.getPreferences().setSoftKeyboardEnabled(true);
                 KeyboardUtils.clearDisableSoftKeyboardFlags(mActivity);
-                KeyboardUtils.showSoftKeyboard(mActivity, mActivity.getTerminalView());
+                mActivity.getTerminalView().postDelayed(getShowSoftKeyboardRunnable(), 300);
```

Comment:
Is this delay necessary? Just requesting focus makes it works for me. I see no difference in behavior with or without the delay, except that toggling the keyboard now has a delay.

Reply:
Yes, as mentioned in the comment above it. It requires a delay after steps from #2112 on my device. It works with 100ms too, but may not work for [other slower devices](https://stackoverflow.com/a/29230447/14686958). I have no way to know the correct one unless multiple testers come and say it works for them, so better to set a looser delay, since releases are slow. And at least `300ms` may be required anyways. The `setSoftKeyboardVisibility()` called by `onFocusChange()` has it at `500ms`.

Reply:
Okay, that's a bit surprising. I see that the delay is required when auto opening the keyboard after returning to Termux (the delay later down in this file), but here there would be some time between returning to Termux and pressing the button anyways (I don't think one would be able to press the button within a couple of hundred ms after returning to the app). I'm not doubting what you're saying, just wanted to make sure it's necessary in both cases.

The problem is not so much that there is a delay after returning to the app though, but that this also causes a delay when you haven't left the app. E.g. just press the toggle to disable and then press again to enable. This delay is not present in master, so I would assume it's not necessary in this case? Would it be possible to differentiate between these cases, so the delay is only used when you return to Termux, and not when you toggle the keyboard when Termux is active all the time?

Reply:
>, but here there would be some time between returning to Termux and pressing the button anyways

I was wondering about that too, I tried multiple configurations, but delays seems necessary.

>Would it be possible to differentiate between these cases

Yeah, a boolean can be added, just like `mShowSoftKeyboardIgnoreOnce`, delays seems to be required only after app changes.

Reply:
> Yeah, a boolean can be added, just like mShowSoftKeyboardIgnoreOnce, delays seems to be required only after app changes.

Great, or alternatively we could call it both immediately and with a delay? That would fix the issue when toggling when Termux is active, and it wouldn't introduce a delay after app switch for devices that don't need it. Or would that be too hacky?

Reply:
In that case, if user hid the keyboard with back button before timer finishes, it would pop up again. I have added the boolean. Kindly test, thanks.

Reply:
True, but since it's only 300 ms I wouldn't think it's an issue in practice.

I tested a bit more and I see that if I press back out of Termux and open it again (not exiting Termux, but so that onDestroy and onCreate is called), I have some delay even if I remove the delay from the code. However if I just switch to another app or home and back (so that only onStop, onStart and onResume is called) there is only the delay added in the code.

This makes me suspect the delay is only necessary if onDestroy/onCreate is called. Could you test if the delay is necessary on your device if you switch to another app and back to Termux (so that you trigger just onStop/onStart/onResume, not onDestroy/onCreate)?

Reply:
> I have some delay

Delay in what? Enabling keyboard with toggle if it was previously disabled?. `onCreate` will always have its own delay to set up stuff, so showing keyboard automatically or with toggle will have "some" delay anyways.

Reply:
Delay between pressing the keyboard toggle and the keyboard appearing. That is after returning to the app when the keyboard is disabled, so not when it appears automatically.

These are the two cases:

With destroy:
1. Press the toggle to disable the keyboard.
2. Press back to leave (but not exit) the app.
3. Return to the app by pressing it in recent apps or the launcher.
4. Press the keyboard toggle to open the keyboard again.
5. The keyboard will appear.

Here there is some delay between 4. and 5. even if I remove the delay from the code.

Without destroy:
1. Press the toggle to disable the keyboard.
2. Press home to go to the home screen.
3. Return to the app by pressing it in recent apps or the launcher.
4. Press the keyboard toggle to open the keyboard again.
5. The keyboard will appear.

Here the only delay between 4. and 5. is the delay specified in the code. If I remove that delay the keyboard opens immediately.

I'm asking if the delay specified in the code is necessary in both cases for the keyboard to appear on your device, or if it is sufficient to have it in the first case.

Reply:
It seems its only required if `onCreate` is called. I have made the changes, but if its also required even when `onResume` is called without `onCreate` on startup on some device, then we all know who will be coming after.

Reply:
Perfect, thanks! Working great for me now.

Reply:
Great, welcome. Merging.